### PR TITLE
Generate jerryscript-config.h based on build options

### PR DESCRIPTION
**Pull request by galpeter**
Previously after the library was build there was no easy way
to get back the information on how it was build.

This change adds a bit of CMake code to generate the
jerryscript-config.h file containing the default options
and adding the modified build options/features.

### CODE REVIEW

**Comment by galpeter** [comment]
This depends on: #2927 and #2933

File: jerry-core/CMakeLists.txt [hunk 1]
```diff
@@ -280,6 +298,88 @@ set(DEFINES_JERRY ${DEFINES_JERRY} JERRY_GLOBAL_HEAP_SIZE=${JERRY_GLOBAL_HEAP_SI
...
+    # first remove "#define" part of the string
+    string(REGEX REPLACE "\r?\n[ ]*#[ ]*define[ ]+" "" DEFINE_KEY_VALUE "${DEFINE_ENTRY}")
+    string(FIND "${DEFINE_KEY_VALUE}" " " DEFINE_KEY_IDX)
+    string(LENGTH "${DEFINE_KEY_VALUE}" DEFINE_LENGTH)
+
+    if (DEFINE_KEY_IDX EQUAL "-1")
+        set(DEFINE_KEY ${DEFINE_KEY_VALUE})
+    else()
+        string(SUBSTRING "${DEFINE_KEY_VALUE}" 0 ${DEFINE_KEY_IDX} DEFINE_KEY)
+        string(SUBSTRING "${DEFINE_KEY_VALUE}" ${DEFINE_KEY_IDX} 64 DEFINE_VALUE)
```
Comment (dbatyai) [inline jerry-core/CMakeLists.txt]: Why is the substring length 64? We could use -1 if we want the remaining string.

File: jerry-core/CMakeLists.txt [hunk 2]
```diff
@@ -280,6 +298,88 @@ set(DEFINES_JERRY ${DEFINES_JERRY} JERRY_GLOBAL_HEAP_SIZE=${JERRY_GLOBAL_HEAP_SI
...
+  # the config default.
+
+  # If the define loaded from the config file have a different value than the
+  # relevant option passed for the CMake means that it does not have a default value.
+  if(DEFINED ${KEY} AND NOT (${KEY} STREQUAL ${VALUE}))
+    set(JERRY_MODIFIED_OPTIONS "${JERRY_MODIFIED_OPTIONS}#define ${KEY} ${${KEY}}\n")
+  endif()
+endforeach()
+
+# Generate the jerryscipr-config.h file into the build directory
```
Comment (dbatyai) [inline jerry-core/CMakeLists.txt]: Typo: jerryscrip**t**-config.h

File: jerry-core/CMakeLists.txt (hunk 1 above)
Comment (galpeter) [reply jerry-core/CMakeLists.txt]: ohh.. cool! thanks that works

**Pull request merged**